### PR TITLE
Ensure that preview-specific errors are written to console

### PR DESCRIPTION
**Pull request by d-ronnqvist**
Bug/issue #, if applicable: rdar://125564508

## Summary

This fixes a bug where preview-specific errors wasn't written to console.

## Dependencies

None

## Testing

The easiest preview-specific error to reproduce is an already used port.

With any documentation, open two terminal windows and start previewing the documentation on some port in the first window.

Now, preview the same documentation on the same port (using `--port`) in the other window. DocC should fail _with_ an error message that the port is already used. 

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- ~[ ] Added tests~ There is no integration test setup that can verify this fix. 
- [x] Ran the `./bin/test` script and it succeeded
- ~[ ] Updated documentation if necessary~

### CODE REVIEW

**Comment by d-ronnqvist** [comment]
@swift-ci please test

File: Sources/SwiftDocCUtilities/Action/Actions/PreviewAction.swift [hunk 1]
```diff
@@ -59,8 +59,6 @@ public final class PreviewAction: Action, RecreatingContext {
     
     /// A unique ID to access the action's preview server.
     let serverIdentifier = ProcessInfo.processInfo.globallyUniqueString
-
-    private let diagnosticEngine: DiagnosticEngine
```
Comment (d-ronnqvist) [inline Sources/SwiftDocCUtilities/Action/Actions/PreviewAction.swift:63]: One bug was that this engine didn't have any consumers, so there was nothing to write the diagnostics that it received.

File: Sources/SwiftDocCUtilities/Action/Actions/PreviewAction.swift [hunk 2]
```diff
@@ -155,7 +151,10 @@ public final class PreviewAction: Action, RecreatingContext {
             try servers[serverIdentifier]!.start()
             previewResult = ActionResult(didEncounterError: false)
         } catch {
+            let diagnosticEngine = convertAction.diagnosticEngine
             diagnosticEngine.emit(.init(description: error.localizedDescription, source: nil))
+            diagnosticEngine.flush()
```
Comment (d-ronnqvist) [inline Sources/SwiftDocCUtilities/Action/Actions/PreviewAction.swift:156]: Another bug was that the preview action emitted a diagnostic and then returned (below) without telling the engine to write the received diagnostics.

**Comment by d-ronnqvist** [comment]
@swift-ci please test

**Review by daniel-grumberg — APPROVED** [review]
LGTM!

**Comment by d-ronnqvist** [comment]
@swift-ci please test

**Pull request merged**